src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_sslstream.c

```diff
@@ -487,21 +493,27 @@ int32_t AndroidCryptoNative_SSLStreamSetTargetHost(SSLStream* sslStream, char* t
     // ArrayList<SNIServerName> nameList = new ArrayList<SNIServerName>();
     // SNIHostName hostName = new SNIHostName(targetHost);
     // nameList.add(hostName);
-    loc[hostStr] = make_java_string(env, targetHost);
-    loc[nameList] = (*env)->NewObject(env, g_ArrayListClass, g_ArrayListCtor);
-    ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
-    loc[hostName] = (*env)->NewObject(env, g_SNIHostName, g_SNIHostNameCtor, loc[hostStr]);
-    ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
-    (*env)->CallBooleanMethod(env, loc[nameList], g_ArrayListAdd, loc[hostName]);
-    ON_EXCEPTION_PRINT_AND_GOTO(cleanup);
+    if (g_SNIHostName != NULL)
```

Comment:
Shouldn't this always be true, since the function is already returning early if `g_SNIHostName == NULL`?